Update README to include more details (#382)
vmware-tanzu/cluster-api-provider-bringyourownhost: README.md

## What is Cluster API Provider BYOH

[Cluster API](https://github.com/kubernetes-sigs/cluster-api) brings
declarative, Kubernetes-style APIs to cluster creation, configuration and
management.

__BYOH__ is a Cluster API Infrastructure Provider for already-provisioned hosts running Linux. This provider allows operators to adopt Cluster API for deploying and managing kubernetes nodes without also having to adopt a specific infrastructure service. This enables users to decouple kubernetes node provisioning from host and infrastructure provisioning.

## BYOH Glossary
**Host** - A host is a running computer system. It could be physical or virtual. It has a kernel and some base operating system

> **BYO Host** - A host provisioned and managed outside of Cluster API

Reviewer: ```suggestion
**BYO Host** - A Linux host provisioned and managed outside of Cluster API
```

## Compatibility with Cluster API and Kubernetes Versions

- BYOH is currently compatible wth Cluster API v1beta1 (v1.0)

> - BYOH installer support is present and tested only for Kubernetes version `v1.22.3`. You may however use it to provision clusters of a different version, but install the Kubernetes components yourself (use the `--skip-installation` flag when starting the agent)

Reviewer: ```suggestion
- BYOH installer support is present and tested only for Kubernetes version `v1.22.3`. You may however use it to provision clusters of a different version by manually installing Kubernetes components (use the `--skip-installation` flag when starting the agent)
```
